### ISSUE TITLE
Ensure openssl s_client -quic failure handling works similar to TLS 

### ISSUE THREAD

**Issue by hlandau** [openssl/project#255]
When using openssl s_client in QUIC mode, the following cases should work similarly to TLS:
- Connecting to wrong host/port
- Connecting to wrong ALPN
- Combinations of these etc.

**Comment by arapov** [comment]
- Understand the reasons for the differences if there are any.
  - we assume that there are some.

**Comment by hlandau** [comment]
>We might want to raise an extra error if we never receive any packet from the peer indicating that there is probably no peer listening. I do not think we can do much more.

Yeah. Currently we don't have any kind of connect timeout so only the idle timeout catches it. We should probably have an explicit connect timeout which is modelled separately.

**Comment by t8m** [comment]
> Yeah. Currently we don't have any kind of connect timeout so only the idle timeout catches it. We should probably have an explicit connect timeout which is modelled separately.

Is this something we need for 3.2.0? Is it a beta blocker?

**Comment by hlandau** [comment]
I suppose since it is just a different error it is more a quality of life thing and not really a beta blocker.

It's not hard, you would just add an OSSL_TIME to QUIC_CHANNEL, set it to ossl_time_infinite() to 'disarm' it when receiving the first encrypted or retry packet, and have some nice specific TCAUSE if it expires. Timeout should probably be slightly less than the idle timeout and checked before it.

**Comment by t8m** [comment]
This is not going to be fixed for 3.2.0 at this point as this is just a minor usability feature/bug. I'll restart working on this after finishing the providers meeting preparation.
